packages/cli/src/lib/builder/DynamicImportPreservePlugin.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Plugin } from 'rollup';
+
+export default function dynamicImportPreservePlugin(): Plugin {
```

Comment:
I was surprised there was nothing available for this wheen looking into it a while ago. Is there still really nothing on NPM or new config in rollup that lets you do this without a custom plugin? :sweat_smile:

Reply:
I really wasn't able to find anything. This came from a thread where someone wanted to do something similar but it wasn't a full plugin.

Reply:
Turns out this is actually supported with a rollup option, https://rollupjs.org/configuration-options/#output-dynamicimportincjs. Adjusting this to use that property.

Reply:
Nice! 👍, thought something like this would be added to rollup eventually 😅